File: CucumberJVM/Chapter13/13.1-TheLocalScenarioContextPattern/After/src/main/java/com/wimp/localscenariocontext/app/services/AuthenticationService.java
package com.wimp.localscenariocontext.app.services;

import com.wimp.localscenariocontext.app.infrastructure.UserSession;

public class AuthenticationService {
    public static void login(String customerName) {
        UserSession.getCurrent().setLoggedInCustomerName(customerName);
    }

    public static boolean isLoggedIn(String customerName) {
        return customerName.equals(UserSession.getCurrent().getLoggedInCustomerName());
    }
}

File: Reqnroll/Chapter13/13.3-IntroduceAndReferPattern/Before/WIMP.App/Services/AuthenticationService.cs
using WIMP.App.Infrastructure;

namespace WIMP.App.Services;

public static class AuthenticationService
{
    public static void Login(string customerName)
    {
        UserSession.Current.LoggedInCustomerName = customerName;
    }

    public static bool IsLoggedIn(string? customerName)
    {
        return customerName != null && UserSession.Current.LoggedInCustomerName == customerName;
    }
}
